torchrec/modules/fused_embedding_modules.py

```diff
@@ -479,6 +479,10 @@ def forward(self, features: KeyedJaggedTensor) -> KeyedTensor:
             if self._is_weighted:
                 weights = torch.cat(weights)
 
+            # code to transfer unsharded model to current_device where table is
+            if emb_op._emb_module.weights_dev.is_cuda:
```

Comment:
This shouldn't be necessary (moving the TBE to torch during forward). something must be wrong

Also, we typically don't want to access weights_dev directly

Reply:
We believe this may be the cause for [deadlock on AMD/NV gpus](https://github.com/pytorch/torchrec/pull/994) (without explicit GPU synchronization, cache flushing) .

[This](https://github.com/pytorch/torchrec/blob/3056d6dd211571d69d89c4c441f1bd2ab0edb9e7/torchrec/distributed/tests/test_fused_embedding_bag_collection.py#L57) is supposed to move the Module on to the right gpu, but `_emb_module` will not move due to it [being registered as a List](https://github.com/pytorch/torchrec/blob/3056d6dd211571d69d89c4c441f1bd2ab0edb9e7/torchrec/modules/fused_embedding_modules.py#L391). [This ](https://github.com/pytorch/torchrec/blob/3056d6dd211571d69d89c4c441f1bd2ab0edb9e7/torchrec/modules/fused_embedding_modules.py#L443)may be the replacement mechanism, but it does not seem to be working.

We suspect that the code still does work sporadically on NV gpus, because of peer accessible memory between gpus (if caches are flushed). In any case, please look at the output that @bmedishe posted in the description. It shows that weights_dev are not on the right device. 

If there is a better mechanism to fix this, please let us know.

Reply:
Regarding `weights_dev` being accessed in this layer of the software, currently, we did not find any other mechanism, since one cannot check where a Module is in torch, and this function does not seem to know `device.type` (used in `__init__`. but invalid afterwards, as `FusedEmbeddingBagCollection` has since migrated)

Reply:
@amathews-amd  -> yes, you're right, this is causing deadlock, this module is breaking internally and we have disabled this internal test. (for Nvidia GPUs)
Let us get back to you with a fix